rustls/src/client/client_conn.rs

```diff
@@ -215,6 +216,15 @@ pub struct ClientConfig {
 
     /// How to verify the server certificate chain.
     pub(super) verifier: Arc<dyn verify::ServerCertVerifier>,
+
+    /// How to decompress the server's certificate chain.
+    ///
+    /// If this is non-empty, the [RFC8779] certificate compression
+    /// extension is offered, and any compressed certificates are
+    /// transparently decompressed during the handshake.
```

Comment:
I think there should be a mention here that the extension is only offered for TLS 1.3:

> This extension is only supported with TLS 1.3 [RFC8446] and newer; if TLS 1.2 [RFC5246] or earlier is negotiated, the peers MUST ignore this extension.

In the ECH PR we tried to make it hard to enable ECH without also enabling TLS1.3 as a protocol version. Should there be some consideration given to that here?

Reply:
I don't think so. One of my goals in this PR is (aside from opting-in via the crate features) avoiding any explicit configuration to have things work in the common case (and, eg, requiring that callers clear out `ClientConfig::cert_decompressors` to use TLS1.2 wouldn't gel with that). I think that is appropriate for what should be treated as an opportunistic performance improvement.

(Whereas with ECH, it is actively harmful to privacy to get the mandatory configuration wrong.)

Reply:
Will add a note to this comment.

Reply:
This makes sense to me :+1: Thanks for adding a comment.